### PR TITLE
add port forwards listing

### PR DESCRIPTION
**Pull request by lucieleblanc**
Creates a new resource that lists all port forwards in use.

### CODE REVIEW

File: list_port_forwards/display_port_forwards.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+echo "\___)  \___)      (___/  (___/"
+
+echo "---- Port forwards in use ----"
+
+jqquery='.items[] | { name:.metadata.annotations."tilt.dev/resource", p:.spec.forwards[] } | "\(.name):\(.p.containerPort) -> \(.p.host):\(.p.localPort)"'
+
+startquote='s/^"//'
+endquote='s/"$//'
+
+tilt get portforwards -o json | jq "$jqquery" | sed -e "$startquote" -e "$endquote"
```
Comment (KOGI) [inline list_port_forwards/display_port_forwards.sh:13]: This assumes the presnce of `jq` on the user's machine...

File: list_port_forwards/display_port_forwards.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
```
Comment (nicks) [inline list_port_forwards/display_port_forwards.sh:1]: bash has tricky error-handling. here's a good explanation:
https://wizardzines.com/comics/bash-errors/

I'd probably add
set -euo pipefail
here, which is closer to the semantics you want

File: list_port_forwards/display_port_forwards.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+echo "__(.)= __(.)=    =(.)__ =(.)__"
+echo "\___)  \___)      (___/  (___/"
+
+echo "---- Port forwards in use ----"
+
+jqquery='.items[] | { name:.metadata.annotations."tilt.dev/resource", p:.spec.forwards[] } | "\(.name):\(.p.containerPort) -> \(.p.host):\(.p.localPort)"'
+
+startquote='s/^"//'
```
Comment (nicks) [inline list_port_forwards/display_port_forwards.sh:10]: i think `jq -r` will omit the quotes for you, so you don't need to do it manually

File: list_port_forwards/display_port_forwards.sh (hunk 1 above)
Comment (nicks) [reply list_port_forwards/display_port_forwards.sh:13]: ya, maybe add a README that mentions that this extension depends on `jq`?

also, add a check at the top like

```
command -v jq || (echo "jq not installed :(" && exit 1)
```

**Review by nicks — COMMENTED** [review]
ooh this is a cool demo on how to use the portforward api

/cc @ellenkorbes might have other ideas on how the dev interaction should work

**Comment by ellenkorbes** [comment]
Question: Can you have a UIButton under the Tiltfile resource? IMHO that'd be the most natural place for this to live.

But that's just my opinion. This is fine as is.

File: list_port_forwards/display_port_forwards.sh (hunk 2 above)
Comment (lucieleblanc) [reply list_port_forwards/display_port_forwards.sh:1]: Thank you for the explanation!

File: list_port_forwards/display_port_forwards.sh (hunk 1 above)
Comment (lucieleblanc) [reply list_port_forwards/display_port_forwards.sh]: Ah, thank you for pointing that out -- I've added the check.

File: list_port_forwards/display_port_forwards.sh (hunk 3 above)
Comment (lucieleblanc) [reply list_port_forwards/display_port_forwards.sh]: Changed. Thank you!

**Review by nicks — APPROVED** [review]

**Pull request merged**